### PR TITLE
fix broken link for owner references in Finalizers

### PR DESCRIPTION
**Pull request by jizusun**
Currently the link is broken in this page 
https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/#owners-labels-finalizers

> Like labels, [owner references](https://kubernetes.io/concepts/overview/working-with-objects/owners-dependents/) describe the relationships between objects in Kubernetes, but are used for a different purpose. When a controller manages objects like Pods, it uses labels to track changes to groups of related objects. For example, when a Job creates one or more Pods, the Job controller applies labels to those pods and tracks changes to any Pods in the cluster with the same label.

### CODE REVIEW

File: content/en/docs/concepts/overview/working-with-objects/finalizers.md [hunk 1]
```diff
@@ -44,8 +44,7 @@ and the controller deletes the volume.
 
 ## Owner references, labels, and finalizers {#owners-labels-finalizers}
 
-Like {{<glossary_tooltip text="labels" term_id="label">}}, [owner references]
-(/docs/concepts/overview/working-with-objects/owners-dependents/)
+Like {{<glossary_tooltip text="labels" term_id="label">}}, [owner references](/docs/concepts/overview/working-with-objects/owners-dependents/)
```
Comment (tengqm) [inline content/en/docs/concepts/overview/working-with-objects/finalizers.md:47]:
```suggestion
Like {{<glossary_tooltip text="labels" term_id="label">}},
[owner references](/docs/concepts/overview/working-with-objects/owners-dependents/)
```

**Comment by natalisucks** [comment]
Hey there @jizusun! Thanks for contributing, it's very appreciated. In order to properly review and merge this PR, you'll need to sign the CLA that's located [here](https://git.k8s.io/community/CLA.md#the-contributor-license-agreement). If you're having trouble, you'll need to verify that your email is set on your git commits so that it can be recognized and we can move forward with your first contribution. Thanks again!

File: content/en/docs/concepts/overview/working-with-objects/finalizers.md (hunk 1 above)
Comment (jizusun) [reply content/en/docs/concepts/overview/working-with-objects/finalizers.md:47]: Hi, do you mean to add a new line break?

**Comment by jizusun** [comment]
> Pls sign the CLA in order for the PR to be reviewed. more info [#29990 (comment)](https://github.com/kubernetes/website/pull/29990#issuecomment-939211094)

@Arhell  CLA signed, thank you so much for the guidance.

**Pull request merged**